include/can/core/freertos_can_dispatch.hpp

```diff
@@ -0,0 +1,91 @@
+#pragma once
+
+#include "FreeRTOS.h"
+#include "can_message_buffer.hpp"
+#include "common/core/freertos_message_buffer.hpp"
+#include "dispatch.hpp"
+#include "message_core.hpp"
+
+using namespace message_core;
+using namespace message_buffer;
+using namespace can_message_buffer;
+using namespace can_dispatch;
+using namespace freertos_message_buffer;
+
+namespace freertos_can_dispatch {
+
+/**
+ * A FreeRTOS task entry point that polls a MessageBuffer.
+ * @tparam BufferType The MessageBuffer type
+ * @tparam Listener The CanMessageBufferListener type
+ */
+template <MessageBuffer BufferType, CanMessageBufferListener Listener>
+class FreeRTOSCanBufferPoller {
+  public:
+    /**
+     * Constructor
+     * @param buffer The MessageBuffer instance (ie FreeRTOSMessageBuffer)
+     * @param listener The listener to be called back when a new message is in
+     * the buffer.
+     */
+    FreeRTOSCanBufferPoller(BufferType& buffer, Listener& listener)
+        : buffer{buffer}, listener{listener}, reader{buffer, listener} {}
+
+    /**
+     * Task entry point.
+     */
+    void operator()() {
+        for (;;) {
+            reader.read(portMAX_DELAY);
+        }
+    }
+
+  private:
+    BufferType& buffer;
+    Listener& listener;
+    CanMessageBufferReader<BufferType, Listener> reader;
+};
+
+/**
+ * A helper to build a FreeRTOMessageBuffer holding can messages to build
+ * supporting classes to poll and parse the messages.
+ *
+ * @tparam BufferSize The size of the FreeRTOMessageBuffer in bytes
+ * @tparam HandlerType The type of the parsed message callback. Must conform to
+ * HandlesMessages concept.
+ * @tparam MessageTypes The CanMessage types this MessageBuffer is interested
+ * in.
+ */
+template <std::size_t BufferSize, typename HandlerType,
+          CanMessage... MessageTypes>
+requires HandlesMessages<HandlerType, MessageTypes...>
```

Comment:
i think we might also want to require `!std::is_move_constructible<HandlerType> && !std::is_move_assignable<HandlerType>` to prevent the handler, which will be captured by a long-lived reference, from getting changed out from under us. We could also require `!std::is_copy_{assignable,constructible}` to prevent people from accidentally making a copy that isn't actually associated with this anymore but that wouldn't be invalid behavior just probably wrong behavior.